tensorflow/compiler/xla/service/gpu/amdgpu_compiler.cc

```diff
@@ -119,14 +119,7 @@ AMDGPUCompiler::AMDGPUCompiler()
                   amdgpu::TargetTriple(), amdgpu::DataLayout()) {}
 
 GpuVersion AMDGPUCompiler::GetGpuVersion(se::StreamExecutor* stream_exec) {
-  std::string gcn_arch_name =
-      stream_exec->GetDeviceDescription().rocm_amdgpu_gcn_arch_name();
-  if (gcn_arch_name == stream_exec->GetDeviceDescription().kUndefinedString) {
-    LOG(WARNING) << "Couldn't get AMDGPU GCN Arch for device; assuming gfx900.";
```

Comment:
Do we know when and why this could happen? In general I'm not keen on ignoring errors with logs.

Reply:
This should not happen at all, in theory. The check here is just something we had in place for a long time...we can get rid of it, or convert to error, if need be